File: 231122/배열 놀이/play-with-array.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n')
const [n, q] = input[0].split(' ').map(ele => Number.parseInt(ele));
const arr = input[1].split(' ').map(ele => Number.parseInt(ele));
for (let i = 2; i < input.length; i++) {
    const query = input[i].split(' ').map(ele => Number.parseInt(ele));
    const qType = query[0];
    if (qType === 1) {
        const order = query[1];
        handleQueryTypeOne(arr, order);
    } else if(qType === 2) {
        const target = query[1];
        handleQueryTypeTwo(arr, target);
    } else if (qType === 3) {
        const from = query[1];
        const to = query[2];
        handleQueryTypeThree(arr, from, to);
    }
}

function handleQueryTypeOne(arr, order) {
    console.log(arr[order-1]);
}

function handleQueryTypeTwo(arr, target) {
    const index = arr.findIndex(ele => ele === target);
    console.log(index === -1 ? 0 : index+1);
}

function handleQueryTypeThree(arr, from, to) {
    console.log(arr.slice(from-1, to).join(' '));
}

File: 231122/배열 놀이/play-with-array.java
import java.util.*;
import java.util.stream.IntStream;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException, NumberFormatException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int q = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        for (int i = 0; i < q; i++) {
            st = new StringTokenizer(br.readLine());
            int qType = Integer.parseInt(st.nextToken());
            if (qType == 1) {
                int order = Integer.parseInt(st.nextToken());
                handleQueryTypeOne(arr, order);
            } else if (qType == 2) {
                int target = Integer.parseInt(st.nextToken());
                handleQueryTypeTwo(arr, target);
            } else if (qType == 3) {
                int from = Integer.parseInt(st.nextToken());
                int to = Integer.parseInt(st.nextToken());
                handleQueryTypeThree(arr, from, to);
            }
        }
    }
    public static void handleQueryTypeOne(int[] arr, int order) {
        System.out.println(arr[order-1]);
    }

    public static void handleQueryTypeTwo(int[] arr, int target) {
        int index = IntStream.range(0, arr.length)
                .filter(i -> arr[i] == target)
                .findFirst()
                .orElse(-1);

        System.out.println(index+1);
    }

    public static void handleQueryTypeThree(int[] arr, int from, int to) {
        StringBuilder sb = new StringBuilder();
        Arrays.stream(arr).skip(from-1).limit(to-from+1).forEach(ele -> sb.append(ele).append(' '));
        System.out.println(sb);
    }
}